src/olcs/abstractsynchronizer.js

```diff
@@ -98,27 +98,32 @@ olcs.AbstractSynchronizer.prototype.orderLayers = function() {
  * @private
  */
 olcs.AbstractSynchronizer.prototype.addLayers_ = function(root) {
-  /** @type {Array.<!ol.layer.Base>} */
-  const fifo = [root];
+  /** @type {Array.<Array.<!ol.layer.Base>>} */
```

Comment:
Using an array of array is disturbing. Could you introduce a new `olcsx.LayerWithParents` type (in olcsx.js) and use it here?

It would look like:
```js
/**
 * @typedef {{
 *   layer: {ol.layer.Layer},
 *   parents: {Array<ol.layer.Group>}
 * }}
 */
olcsx.LayerWithParents;
```